src/main/java/seedu/address/model/travelplanner/Directory.java

```diff
@@ -1,16 +1,12 @@
 package seedu.address.model.travelplanner;
 
+
+import javafx.collections.ObservableList;
+import seedu.address.model.activity.Activity;
+
 /**
  * Represents the current directory of wanderlust, which can be instance of TravelPlan or Wishlist
  */
 public abstract class Directory {
-    /**
-     * Returns true if current directory is a travel plan, false otherwise.
-     */
-    public abstract boolean isTravelPlan();
-
-    /**
-     * Returns true if current directory is a wishlist, false otherwise.
-     */
-    public abstract boolean isWishlist();
+    public abstract ObservableList<Activity> getActivityList();
```

Comment:
We just need one method from Directory only? What about AccommodationList all that?

Reply:
We can typecast back to Travelplan to access AccommationList, if we are sure the directory is a travelplan.